runtime/src/accounts_index.rs

```diff
@@ -181,9 +187,20 @@ impl<'a, T: 'static + Clone> Iterator for AccountsIndexIterator<'a, T> {
     }
 }
 
+#[derive(Debug, Default)]
+pub struct MintIndex {
+    index: DashMap<Pubkey, MintIndexEntry>,
+    // Map from pubkey to the mint it's stored in, used for cleanup.
+    // Alternative is to store Option<Pubkey> in each AccountInfo in the
+    // AccountsIndex if something is an SPL account with a mint, but then
+    // every AccountInfo would have to allocate `Option<Pubkey>`
+    accounts_to_mints: DashMap<Pubkey, DashSet<Pubkey>>,
```

Comment:
A token account can only be associated with one mint at a time. Do we need the DashSet here?

Reply:
As discussed over DM, we do need a DashSet, as this spans all updates, not a particular point in time.

Reply:
yeah, I think this is true; @carllin how about briefly commenting about this in the source code?